presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java

```diff
@@ -268,15 +268,24 @@ public static void mergeRegressionState(RegressionState state, RegressionState o
     public static String generateAggregationName(String baseName, TypeSignature outputType, List<TypeSignature> inputTypes)
     {
         StringBuilder sb = new StringBuilder();
-        sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, outputType.toString()));
+        sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, getAbbreviatedTypeName(outputType)));
         for (TypeSignature inputType : inputTypes) {
-            sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, inputType.toString()));
+            sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, getAbbreviatedTypeName(inputType)));
         }
         sb.append(CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, baseName.toLowerCase(ENGLISH)));
 
         return sb.toString();
     }
 
+    private static String getAbbreviatedTypeName(TypeSignature type)
+    {
+        String typeName = type.toString();
+        if (typeName.length() > 10) {
+            return typeName.substring(0, 10);
```

Comment:
Should we consider appending a hash of the full name? IIRC we used to do this where the class would look like MyAggSum$xvTG9dC or something. Makes it easier to identify unique patterns when debugging.

Reply:
Our bytecode gen library uses an `AtomicLong` to guarantee uniqueness. https://github.com/prestodb/presto/pull/15711